packages/medusa/src/services/sales-channel.ts

```diff
@@ -73,8 +74,24 @@ class SalesChannelService extends TransactionBaseService<SalesChannelService> {
     throw new Error("Method not implemented.")
   }
 
+  /**
+   * Creates a SalesChannel
+   */
   async create(data: CreateSalesChannelInput): Promise<SalesChannel> {
-    throw new Error("Method not implemented.")
+    return await this.atomicPhase_(async (manager) => {
+      try {
+        const salesChannelRepo: SalesChannelRepository =
+          manager.getCustomRepository(this.salesChannelRepository_)
+
+        const salesChannel = salesChannelRepo.create(data)
+        return await salesChannelRepo.save(salesChannel)
+      } catch (err) {
+        if (err.code === "23505") {
+          throw new MedusaError(MedusaError.Types.DUPLICATE_ERROR, err.detail)
+        }
+        throw err
+      }
+    })
```

Comment:
suggestion(non-blocking): Would it make sense to move the error handling to the built-in handler of `atomicPhase_`? 

Would be something along the lines of:
```ts
async create(data: CreateSalesChannelInput): Promise<SalesChannel> {
    return await this.atomicPhase_(
      async (manager) => {
        const salesChannelRepo: SalesChannelRepository =
          manager.getCustomRepository(this.salesChannelRepository_)

        const salesChannel = salesChannelRepo.create(data)
        return await salesChannelRepo.save(salesChannel)
      },
      async (err: { code: string }) => {
        if (err.code === PostgresError.DUPLICATE_ERROR) {
          throw new MedusaError(
            MedusaError.Types.DUPLICATE_ERROR,
            `some message`
          )
        }
      }
    )
  }
```